### PR TITLE
LPD-26663 Questions are not returned when a space is used in a tag

### PR DESCRIPTION
**Pull request by Mikellorza**
# What is this trying to solve?

[Link to ticket](https://liferay.atlassian.net/browse/LPD-26663)

This PR fix the bug of the LPD-26663 and add question web module to manage questions tests.

# How am I fixing it?

This PR fix the bug of the LPD-26663 and add question web module to manage questions tests. Now we can use tags with spaces and navigate through them in questions application.

# How can you verify that it works?

The PR includes integration tests.

The PR can also be tested through the following steps :

1. Add a Questions widget to the page
2. Create a new Topic
3. In the new Topic add a new question
4. Add a tag with a space in it for example: “Test Tag“
5. Post the question
6. Go to Tags
7. Click on the tag

Actual Result: No question is returned, as upon clicking on it shows “test-tag“
Expected Result: The question would be returned

### CODE REVIEW

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/Link.es.js [hunk 1]
```diff
@@ -9,5 +9,9 @@ import {Link as RouterLink} from 'react-router-dom';
 import {stringToSlug} from '../utils/utils.es';
 
 export default function Link(props) {
+	if (!props.slugTo) {
```
Comment (DiegoHu97) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/Link.es.js:12]: We can use a ternary condition here, like this is more clear I think:  
`return props.slugTo ? ( <RouterLink {...props} to={stringToSlug(props.to)} /> ) : ( <RouterLink {...props} to={props.to} /> );`

File: modules/test/playwright/tests/questions-web/pages/QuestionsPage.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	async goto() {}
+
+	async addNewQuestion(
+		questionBody: string,
+		questionTitle: string,
+		tagName: string
+	) {
+		await this.goto();
+		await this.page
+			.locator('li')
```
Comment (DiegoHu97) [inline modules/test/playwright/tests/questions-web/pages/QuestionsPage.ts:26]: We can simplify this locator like this `this.page.getByRole('button', { name: 'Ask Question' })`  
 `.click();`

File: modules/test/playwright/tests/questions-web/test.properties [hunk 3]
```diff
@@ -0,0 +1,5 @@
+##
+## Testray
+##
+
+    testray.main.component.name= Liferay Questions Web
```
Comment (DiegoHu97) [inline modules/test/playwright/tests/questions-web/test.properties:5]: There is a source format issue here

**Review by DiegoHu97 — CHANGES_REQUESTED** [review]
Hey , I just left a couple of minor changes.

**Comment by Mikellorza** [comment]
ci:test:sf

**Comment by Mikellorza** [comment]
Hey @DiegoHu97 ,

I 've implemented all things you proposed. I've just run the test:sf again and if you see all good we can move forward.

thanks!

**Comment by DiegoHu97** [comment]
ci:forward

**Comment by Mikellorza** [comment]
ci:forward

**Comment by DiegoHu97** [comment]
ci:report:2573169710

**Comment by Mikellorza** [comment]
ci:forward

**Pull request closed without merging**